### DISCUSSION TITLE
Cache HITs with query parameters?

### DISCUSSION THREAD

**Discussion by ay8s** [vercel/next.js#12736] [Help]
We have a F1 Race Calendar built with Next.js and hosted on Vercel. It's my first Next.js project and I'm sure there's a bunch of optimization that could be done to improve the calendar generation and caching. The site is open source over at https://github.com/sportstimes/f1.

Onto my question...

Users can download the calendar after selecting a bunch of options [here](https://f1calendar.com/generate), with all of these options there's quite a large number of url combinations generated. The URL for Google Calendar also appends a timestamp. I need to dig into why this was added on the old site to determine if it's still required. 

I've added a Cache-Control header (in a new PR) but obviously it doesn't serve a cached ics file if a timestamp is appended until the second request. Is there a way around this? Or is there a better solution?

Thanks

**Comment by ay8s** [comment]
So I have Cache-Control set with a s-maxage. However there are many users who have a calendar URL that contains a unique timestamp query parameter. This was implemented years ago to resolve a bug with Google Calendar. 

Is there a way to set a cache key for all of these requests to be pooled and return a single cached response even with the query parameter to retain the bug fix? vs me redirecting users to the same url minus the query param? 

I need to book up on all of the new Static Generation features etc. 

Thanks for any thoughts :)

**Reply by samsends** [reply]
Bumping this please get back

**Reply by olpeh** [reply]
Yes, we would need this as well. I'm a bit shocked that there seems to be no way to achieve this. The default should be the opposite like CloudFront does: you need to specifically allow some query parameters to be used as cache keys. 
This allows for an easy DDoS attack to any service hosted (directly) in Vercel.
